pyfar/classes/transmission_matrix.py

```diff
@@ -0,0 +1,610 @@
+
+        This means the impedance is connected in series with a potential load
+        impedance. See Equation (2-8) in Table I of Reference [1]_.
+
+        Parameters
+        ----------
+        impedance : FrequencyData
+            The impedance data of the series impedance.
+        abcd_cshape : tuple | int, optional
+            Shape of additional dimensions besides (2,2,num_bins).
+            Per default, no additional dimensions are created.
+
+        Returns
+        -------
+        cls : TransmissionMatrix
+            A TransmissionMatrix representing a series impedance.
+
+        """
+
+        if impedance.cshape != (1,):
+            raise ValueError("Number of channels for 'impedance' must be 1.")
+        Z = impedance.freq[0]
+        ones, zeros = np.ones_like(Z), np.zeros_like(Z)
+        series_Z_abcd = FrequencyData([[ones, Z],[zeros, ones]],
+                                      impedance.frequencies)
+        return cls._create_abcd_matrix_broadcast(series_Z_abcd, abcd_cshape)
+
+    @classmethod
+    def create_shunt_admittance(cls, admittance: FrequencyData,
+                                abcd_cshape = ()):
+        """Creates a transmission matrix representing a shunt admittance
+        (parallel connection).
+
+        In this case, the impedance (= 1 / admittance) is connected in parallel
+        with a potential load impedance.
+        See Equation (2-9) in Table I of Reference [1]_.
+
+        Parameters
+        ----------
+        admittance : FrequencyData
+            The admittance data of the element connected in parallel.
+        abcd_cshape : tuple | int, optional
+            Shape of additional dimensions besides (2,2,num_bins).
+            Per default, no additional dimensions are created.
+
+        Returns
+        -------
+        cls : TransmissionMatrix
+            A TransmissionMatrix representing a parallel connection.
+
+        """
+        if admittance.cshape != (1,):
+            raise ValueError("Number of channels for 'admittance' must be 1.")
+        Y = admittance.freq[0]
+        ones, zeros = np.ones_like(Y), np.zeros_like(Y)
+        series_Z_abcd = FrequencyData([[ones, zeros],[Y, ones]],
+                                      admittance.frequencies)
+        return cls._create_abcd_matrix_broadcast(series_Z_abcd, abcd_cshape)
+
+    @staticmethod
+    def create_transformer(transducer_constant : complex) -> np.ndarray:
+        """Creates a frequency-independent transmission matrix representing a
+        transformer.
+
+        See Equation (2-12) in Table I of Reference [1]_.
+
+        Parameters
+        ----------
+        transducer_constant : scalar
+            The transmission ratio with respect to voltage-like quantity,
+            i.e. Uout/Uin.
+
+        Returns
+        -------
+        tmat : np.ndarray
+            A 2x2 array representing the transmission matrix of the transformer
+            and can be cascaded with TransmissionMatrix objects.
```

Comment:
Doesn't it need to be of shape (2,2,1) to be multiplied by FrequencyData objects?

Reply:
No, since this is just an np.array and not a FrequencyData object, this is the correct approach.